internal/controller/certificates/policies/policies.go

```diff
@@ -96,14 +96,16 @@ func NewReadinessPolicyChain(c clock.Clock) Chain {
 // correctness of metadata and output formats of Certificate's Secrets.
 func NewSecretPostIssuancePolicyChain(ownerRefEnabled bool, fieldManager string) Chain {
 	return Chain{
-		SecretTemplateMismatchesSecret,
-		SecretTemplateMismatchesSecretManagedFields(fieldManager),
```

Comment:
Previously, the `cert-manager.io` annotations were just being ignored while checking the annotations in the template.
This PR adds checks that make sure the correct `cert-manager.io` annotations are present and that they have the correct value.
Same goes for the `controller.cert-manager.io/fao` label.